flax/linen/module.py

```diff
@@ -59,8 +61,25 @@
                  {'make_rng': lambda self, name: jax.random.PRNGKey(0)})
 
 
+
 # pylint: disable=protected-access,attribute-defined-outside-init
 
+@dataclasses.dataclass
+class _ShapeDtype:
+  shape: Tuple[int, ...]
+  dtype: Any
+
+  def __repr__(self):
```

Comment:
This __repr__ is essentially a render fn

Reply:
Resolve via the `_ValueRepresentation` API.